tasks/eva_2722/prepare_covid19_processed_analysis_file.py

```diff
@@ -0,0 +1,101 @@
+
+    with open(target_file, 'a') as f:
+        offset = 0
+        limit = batch_size
+        while offset < total_analysis:
+            analysis_from_ena = get_analysis_from_ena(project, offset, limit)
+            for analysis in analysis_from_ena:
+                if analysis['run_ref'] in processed_analysis:
+                    logger.info(f"Analysis file {analysis['run_ref']} found in already processed list")
+                    f.write(f"{analysis['analysis_accession']},{analysis['submitted_ftp']}\n")
+            offset = offset + limit
+
+
+def total_analysis_in_project(project):
+    count_url = f"https://www.ebi.ac.uk/ena/portal/api/filereportcount?accession={project}&result=analysis"
+    response = requests.get(count_url)
+    if response.status_code != 200:
+        logger.error(f"Error fetching total analysis count for project {project}")
+        response.raise_for_status()
+    return response.json()
+
+
+def get_processed_analysis_files(processed_file_directory):
+    processed_analysis = set()
+    for file in os.listdir(processed_file_directory):
+        file_path = os.path.join(processed_file_directory, file)
+        if os.path.islink(file_path):
+            org_path = os.readlink(file_path)
+            logger.info(f"{file_path} is a symlink, reading files from original path {org_path}")
+            processed_analysis.update(get_processed_analysis_files(org_path))
+        elif os.path.isdir(file_path):
+            logger.info(f"{file_path} is a directory, reading files from sub-directories")
+            processed_analysis.update(get_processed_analysis_files(file_path))
+        else:
+            if "_filtered_vcf.gz" in file:
+                file_name = file.split("_")[0]
+                processed_analysis.add(file_name)
+
+    return processed_analysis
+
+
+def get_analysis_from_ena(project, offset, limit):
+    analysis_url = f"https://www.ebi.ac.uk/ena/portal/api/filereport?result=analysis&accession={project}&offset={offset}" \
+                   f"&limit={limit}&format=json&fields=run_ref,analysis_accession,submitted_ftp"
+    response = requests.get(analysis_url)
+    if response.status_code != 200:
+        logger.error(f"Error fetching analysis info from ENA for {project}")
+        response.raise_for_status()
+    return response.json()
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Download analysis for processing from the Covid-19 DP project',
+                                     formatter_class=argparse.RawTextHelpFormatter, add_help=False)
+    parser.add_argument("--project", default='PRJEB45554', required=False,
+                        help="project from which analysis needs to be downloaded")
+    parser.add_argument("--batch-size", default=10000, required=False, help="batch size of ENA analysis download")
+    parser.add_argument("--processed-file-directory", default="/nfs/production3/eva/data/PRJEB45554/30_eva_valid/",
```

Comment:
Since this is a public repo, suggest not using actual cluster file paths...

Reply:
removed default and made the parameter mandatory